SIDFactoryII/source/runtime/editor/components/component_track.cpp

```diff
@@ -1881,6 +1883,40 @@ namespace Editor
 		return m_EventPos;
 	}
 
+	// TODO: deduplicate code, this is almost the same as DoInsertFirstFreeSequence
```

Comment:
Maybe parameterize the function? Only difference is the function that gets the sequence index.

Reply:
Agree

Reply:
Done. In the other branch..